frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -702,43 +696,41 @@ private void putVulnerabilitiesInRequestComponent(RenderRequest request, String
         } else {
             vuls = vulClient.getVulnerabilitiesByComponentIdWithoutIncorrect(componentId, user);
         }
-        request.setAttribute(VULNERABILITY_LIST, vuls);
 
-        putVulnerabilityMetadatasInRequest(request, vuls);
+        putVulnerabilitiesInRequest(request, vuls, user);
+    }
 
+    private void putVulnerabilitiesInRequest(RenderRequest request, List<VulnerabilityDTO> vuls, User user) {
+        CommonVulnerabilityPortletUtils.putLatestVulnerabilitiesInRequest(request, vuls, user);
+        CommonVulnerabilityPortletUtils.putMatchedByHistogramInRequest(request, vuls);
+        putVulnerabilityMetadatasInRequest(request, vuls);
     }
 
     private void addToVulnerabilityVerifications(Map<String, Map<String, VerificationState>> vulnerabilityVerifications,
                                                  Map<String, Map<String, String>> vulnerabilityTooltips,
                                                  VulnerabilityDTO vulnerability){
         String vulnerabilityId = vulnerability.getExternalId();
         String releaseId = vulnerability.getIntReleaseId();
-        if(! vulnerabilityVerifications.containsKey(vulnerabilityId)){
-            vulnerabilityVerifications.put(vulnerabilityId, new HashMap<>());
-        }
-        if(! vulnerabilityTooltips.containsKey(vulnerabilityId)){
-            vulnerabilityTooltips.put(vulnerabilityId, new HashMap<>());
-        }
+        Map<String, VerificationState> vulnerabilityVerification = vulnerabilityVerifications.computeIfAbsent(vulnerabilityId, k -> new HashMap<>());
```

Comment:
I don't get this change. You create a new map instead of updating the existing one, but then the new map is never used except for putting more stuff in it. What's the point?

Same goes for the next line.

Reply:
No, I create a new map only  if it did not exist yet (if the key is not present), and then put it in the super-map. If the key is present, the existing map is returned.
The method `computeIfAbsent` does exactly what was there before, but in a single line of code, which I find more readable.

Reply:
My bad. I was inattentive at reading. You're absolutely right. It all makes sense.